java-persistence/src/main/java/com/whitecloak/training/inventory/gateway/CategoryGateway.java

```diff
@@ -0,0 +1,10 @@
+package com.whitecloak.training.inventory.gateway;
+
+import com.whitecloak.training.inventory.model.Category;
+import com.whitecloak.training.inventory.persistence.entity.ItemEntity;
+
+import java.util.Set;
+
+public interface CategoryGateway {
+    Category create(String categoryName, Set<ItemEntity> itemEntity);
```

Comment:
Entity has been leaked outside the gateway.

Gateway methods should not accept nor return entity object. Uses of such objects should stay inside the gateway, not exposed in any way outside.